pkg/spdk/engine.go

```diff
@@ -1442,11 +1415,14 @@ func (e *Engine) snapshotOperation(spdkClient *spdkclient.Client, inputSnapshotN
 
 func (e *Engine) getReplicaClients() (replicaClients map[string]*client.SPDKClient, err error) {
 	replicaClients = map[string]*client.SPDKClient{}
-	for replicaName := range e.ReplicaAddressMap {
-		if e.ReplicaModeMap[replicaName] == types.ModeERR {
+	for replicaName, replicaStatus := range e.ReplicaStatusMap {
+		if replicaStatus.Mode != types.ModeRW && replicaStatus.Mode != types.ModeWO {
```

Comment:
Slight logic change here in the case that Mode is unset and is "".  But perhaps better, because it will be skipped, just as for ModeERR.